test/integration/package_int_test.go

```diff
@@ -313,29 +314,37 @@ func (suite *PackageIntegrationTestSuite) TestPackage_operation() {
 	username := ts.CreateNewUser()
 	namespace := fmt.Sprintf("%s/%s", username, "python3-pkgtest")
 
-	cp := ts.Spawn("fork", "ActiveState-CLI/Python3", "--org", username, "--name", "python3-pkgtest")
+	cp := ts.Spawn("fork", "ActiveState-CLI/Revert", "--org", username, "--name", "python3-pkgtest")
 	cp.ExpectExitCode(0)
 
 	cp = ts.Spawn("activate", namespace, "--path="+ts.Dirs.Work, "--output=json")
 	cp.ExpectExitCode(0)
 
-	suite.Run("packages add", func() {
-		cp := ts.Spawn("packages", "add", "dateparser@0.7.2")
-		cp.ExpectRe("(?:Package added|The project is currently building)")
-		cp.Wait()
-	})
+	cp = ts.Spawn("history")
+	cp.ExpectExitCode(0)
 
-	suite.Run("packages update", func() {
-		cp := ts.Spawn("packages", "update", "dateparser@0.7.6")
-		cp.ExpectRe("(?:Package updated|The project is currently building)")
-		cp.Wait()
-	})
+	commitRe := regexp.MustCompile(`[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}`)
+	firstCommit := commitRe.FindString(cp.TrimmedSnapshot())
 
-	suite.Run("packages remove", func() {
-		cp := ts.Spawn("packages", "remove", "dateparser")
-		cp.ExpectRe("(?:Package removed|The project is currently building)")
-		cp.Wait()
-	})
+	cp = ts.Spawn("packages", "add", "dateparser@0.7.2")
+	cp.ExpectRe("(?:Package added|The project is currently building)")
+	cp.Wait()
+
+	// cp = ts.Spawn("packages", "update", "dateparser@0.7.6")
+	// cp.ExpectRe("(?:Package updated|The project is currently building)")
+	// cp.Wait()
+
+	// cp = ts.Spawn("packages", "remove", "dateparser")
+	// cp.ExpectRe("(?:Package removed|The project is currently building)")
+	// cp.Wait()
+
+	cp = ts.Spawn("revert", firstCommit)
+	cp.SendLine("y")
+	cp.ExpectExitCode(0)
+
+	cp = ts.Spawn("history")
+	cp.ExpectExitCode(0)
+	cp.Expect("Description: Reverting to commit")
```

Comment:
I believe you will have to switch these two lines.

When you wait for the exit code first, all the output will have been parsed already.